qiskit/tools/visualization.py

```diff
@@ -22,10 +22,14 @@
 import itertools
 import operator
 import re
+import os
+import subprocess
+import shutil
+import logging
 from collections import Counter, OrderedDict
 from functools import reduce
 from io import StringIO
-
+from PIL import Image, ImageChops
```

Comment:
Following up on the idea of making the feature "optional": can we also make `pillow` optional, as it will not be used unless the non-python dependencies are not installed? This would involve:
* making this import "conditional-ish", either using something similar to [how projectq is handled](https://github.com/QISKit/qiskit-sdk-py/blob/master/qiskit/backends/_projectq_simulator.py#L32), or importing it inside the function that uses (they are currently two, but with a bit of moving things around it seems it might be simple to reduce it to one)? 
* remove it from `requirements.txt` (ideally, [marking is as an "extra" dependency](http://setuptools.readthedocs.io/en/latest/setuptools.html#declaring-extras-optional-features-with-their-own-dependencies) in `setup.py.in` - although we are not really using the `extras_require` officially, it might be a good time to also add `projectq` to that dict if you feel inclined to do so)
* update the docstrings and the `try` block to be aware of pillow

Reply:
I implemented your 1st and 3rd bullets.
Regarding the 2nd - I agree with you that pillow is only required for circuit drawing, similar to the other dependencies introduced here. But adding it as `extras_require` would mean that it does not get installed automatically. I think this is not a heavy dependency (unlike `pdflatex` and `poppler`); so wouldn't it be better to remove the burden of manual installation on the user? I also imagine `pillow` will be used in any eventual, light-weight solution for circuit drawing (as indicated in Ismail's original issue #235).
For projectq, I agree -- it is tangential to the SDK. Added to `extras_require`